Public/Src/Cache/ContentStore/Distributed/NuCache/ContentLocationDatabase.cs

```diff
@@ -616,15 +627,26 @@ private Task ForceCacheFlushAsync(OperationContext context)
                 Tracer,
                 async () =>
                 {
+                    long flushedEntries = 0;
                     try
                     {
+                        // Make it likely that this runs in a separate thread other than the one that triggered the
+                        // flush
                         await Task.Yield();
-                        return Result.Success(await _inMemoryCache.FlushAsync(context));
+
+                        var counters = await _inMemoryCache.FlushAsync(context);
+                        flushedEntries = counters[FlushableCache.FlushableCacheCounters.Persisted].Value;
+                        return Result.Success(counters);
                     }
                     finally
                     {
-                        Interlocked.Exchange(ref _cacheUpdatesSinceLastFlush, 0);
+                        Interlocked.Add(ref _cacheUpdatesSinceLastFlush, -flushedEntries);
                         ResetFlushTimer();
+
+                        if (counter != null)
+                        {
+                            Counters[counter.Value].Increment();

```

Comment:
should it be 'counters[counter.Value]' instead?

Reply:
No, Counters is for the ContentLocationDatabase, counters is for the flush. I renamed the flush ones to make it less ambiguous

---
In reply to: [333121903](https://github.com/microsoft/BuildXL/pull/1021#discussion_r333121903) [](ancestors = 333121903)